library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/Ac3Reader.java

```diff
@@ -188,18 +192,36 @@ private boolean skipToNextSync(ParsableByteArray pesBuffer) {
   private void parseHeader() {
     headerScratchBits.setPosition(0);
     Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
-    if (format == null || frameInfo.channelCount != format.channelCount
-        || frameInfo.sampleRate != format.sampleRate
-        || frameInfo.mimeType != format.sampleMimeType) {
+    if ((format == null || frameInfo.sampleRate != format.sampleRate
+        || frameInfo.mimeType != format.sampleMimeType)
```

Comment:
Thanks for all the information! I think we'd like to accept all of the changes, but we'd also like to think about this particular change in a bit more detail before we do, and we'll need to find some time for that :).

It would be good to split the changes to the other two files into a separate pull request, just so that we can get those changes merged in rather than them being held up whilst we do this. If you could just leave this pull request intact, so this discussion thread doesn't disappear anywhere, that would be great, and we'll follow up here regarding this change once we've mulled it over a bit more.

Thanks for your help!